### DISCUSSION TITLE
VMware to GCP issue

### DISCUSSION THREAD

**Discussion by kkbaum** [cisagov/Malcolm#510] [Troubleshooting] [answered]
I have a working MALCOLM image from VMWare Workstation that our network engineer converted and uploaded to our Google Cloud platform. When starting MALCOLM I get the following errors constantly till I kill the script: 

pcap-monitor-1        | 2024-11-22 18:49:50 WARNING: GET http://opensearch:9200/_cluster/health [status:N/A request:4.345s]
pcap-monitor-1        | ConnectionRefusedError: [Errno 111] Connection refused
pcap-monitor-1        |
pcap-monitor-1        | The above exception was the direct cause of the following exception:
pcap-monitor-1        |
pcap-monitor-1        | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f6062989f50>: Failed to establish a new connection: [Errno 111] Connection refused

I know Google makes changes to accommodate the image but I have yet in three days discovered what could be causing the issue. I have rebuilt MALCOLM from the Git files, verified network settings, turned off any firewalls and even changed the MTU of the Docker0 interface to match the local adapter.

**Comment by mmguero** [comment]
It's normal to see that maybe once or twice during startup, but eventually once opensearch starts it should resolve. The most likely issues is a lack of resources for the OpenSearch container. So, a couple of questions for you:

1. What are the specs for the VM as far as memory/CPU goes?
2. What's the output of `grep JAVA_OPTS config/*.env` when run from your Malcolm installation directory?

**Comment by kkbaum** [comment]
1. Specs are an e2-medium which consists of 2 single-core vCPUs  and 4gb RAM.
2. Output from files: 
config/logstash.env:LS_JAVA_OPTS=-server -Xmx2g -Xms2g -Xss1536k -XX:-HeapDumpOnOutOfMemoryError -Djava.security.egd=file:/dev/./urandom -Dlog4j.formatMsgNoLookups=true
config/opensearch.env:OPENSEARCH_JAVA_OPTS=-server -Xmx3500m -Xms3500m -Xss256k -XX:-HeapDumpOnOutOfMemoryError -Djava.security.egd=file:/dev/./urandom -Dlog4j.formatMsgNoLookups=true

**Reply by mmguero** [reply, marked as answer]
That's your issue, then. Malcolm [requires at a minimum](https://cisagov.github.io/Malcolm/docs/system-requirements.html#SystemRequirements) 16GB of ram and 8 logical CPUs.
